service/src/main/java/com/zebrunner/reporting/service/feign/FeignClientInterceptor.java

```diff
@@ -0,0 +1,25 @@
+package com.zebrunner.reporting.service.feign;
+
+import com.zebrunner.reporting.domain.dto.auth.AuthenticatedUser;
+import feign.RequestInterceptor;
+import feign.RequestTemplate;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FeignClientInterceptor implements RequestInterceptor {
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String TOKEN_TYPE = "Bearer";
+
+    @Override
+    public void apply(RequestTemplate requestTemplate) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication != null && authentication.getPrincipal() instanceof AuthenticatedUser) {
+            AuthenticatedUser authenticatedUser = (AuthenticatedUser) authentication.getPrincipal();
+            requestTemplate.header(AUTHORIZATION_HEADER, String.format("%s %s", TOKEN_TYPE, authenticatedUser.getToken()));
+        }
+    }
```

Comment:
Add empty line before closing class bracket, add new line to the end of the file